clients/include/testing_herk.hpp

```diff
@@ -0,0 +1,275 @@
+#include "flops.hpp"
+#include "near.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T>
+void testing_herk_bad_arg(const Arguments& arg)
+{
+    const rocblas_int N = 100;
+    const rocblas_int K = 100;
+
+    const rocblas_int lda = 100;
+    const rocblas_int ldc = 100;
+
+    using U       = real_t<T>;
+    const U alpha = 1.0;
+    const U beta  = 1.0;
+
+    const size_t            safe_size = 100;
+    const rocblas_fill      uplo      = rocblas_fill_upper;
+    const rocblas_operation transA    = rocblas_operation_none;
+
+    rocblas_local_handle handle;
+
+    // allocate memory on device
+    device_vector<T> dA(safe_size);
+    device_vector<T> dC(safe_size);
+    if(!dA || !dC)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T, U>)(nullptr, uplo, transA, N, K, &alpha, dA, lda, &beta, dC, ldc),
+        rocblas_status_invalid_handle);
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T,
+                      U>)(handle, rocblas_fill_full, transA, N, K, &alpha, dA, lda, &beta, dC, ldc),
+        rocblas_status_invalid_value);
+
+    EXPECT_ROCBLAS_STATUS((rocblas_herk<T, U>)(handle,
+                                               uplo,
+                                               rocblas_operation_transpose,
+                                               N,
+                                               K,
+                                               &alpha,
+                                               dA,
+                                               lda,
+                                               &beta,
+                                               dC,
+                                               ldc),
+                          rocblas_status_invalid_value);
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T, U>)(handle, uplo, transA, N, K, nullptr, dA, lda, &beta, dC, ldc),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T, U>)(handle, uplo, transA, N, K, &alpha, nullptr, lda, &beta, dC, ldc),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T, U>)(handle, uplo, transA, N, K, &alpha, dA, lda, nullptr, dC, ldc),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T, U>)(handle, uplo, transA, N, K, &alpha, dA, lda, &beta, nullptr, ldc),
+        rocblas_status_invalid_pointer);
+
+    // quick return with invalid pointers
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_herk<T,
+                      U>)(handle, uplo, transA, 0, K, nullptr, nullptr, lda, nullptr, nullptr, ldc),
+        rocblas_status_success);
```

Comment:
Subtle difference, but the nullptr can definitely be considered bad arg.    "special" arguments is the purpose here.    Could likely expand to cover all quick return cases, otherwise we repeat the same tests over and over if included in yaml sets.